allennlp/interpret/attackers/hotflip.py

```diff
@@ -0,0 +1,203 @@
+                elmo_tokens = []
+                indexer = self.predictor._dataset_reader._token_indexers["elmo"]
+                for tok in all_tokens:
+                    tmp = indexer.tokens_to_indices([Token(text=tok)], self.vocab, "sentence")["sentence"]
+                    elmo_tokens.append(tmp[0])
+                all_inputs["elmo"] = torch.LongTensor(elmo_tokens).unsqueeze(0)
+
+        # find the TextFieldEmbedder
+        for module in self.predictor._model.modules():
+            if isinstance(module, TextFieldEmbedder):
+                embedder = module
+        # pass all tokens through the fake matrix and create an embedding out of it.
+        embedding_matrix = embedder(all_inputs).squeeze()
+        return Embedding(num_embeddings=self.vocab.get_vocab_size('tokens'),
+                         embedding_dim=embedding_matrix.shape[1],
+                         weight=embedding_matrix,
+                         trainable=False)
+
+    def attack_from_json(self,
+                         inputs: JsonDict = None,
+                         name_of_input_field_to_attack: str = 'tokens',
+                         name_of_grad_input_field: str = 'grad_input_1',
+                         ignore_tokens: List[str] = ["@@NULL@@"]): # pylint disable=unused-argument,dangerous-default-value
+
+        """
+        Replaces one token at a time from the input until the model's prediction changes.
+        `name_of_input_field_to_attack` is for example `tokens`, it says what the input
+        field is called. `name_of_grad_input_field` is for example `grad_input_1`, which
+        is a key into a grads dictionary.
+
+        The method computes the gradient w.r.t. the tokens, finds
+        the token with the maximum gradient (by L2 norm), and replaces it will
+        another token based on the first-order Taylor approximation of the loss.
+        This process is iteratively repeated until the prediction changes.
+        Once a token is replaced, it is not flipped again.
+
+        TODO (@Eric-Wallace) add functionality for ignore_tokens in the future.
+        """
+        original_instances = self.predictor.inputs_to_labeled_instances(inputs)
+        original_tokens = list(original_instances[0][name_of_input_field_to_attack].tokens)
+        final_tokens = []
+        new_prediction = None
+        for new_instances in original_instances:
+            new_instances = [new_instances]
+            test_instances = self.predictor.inputs_to_labeled_instances(inputs)
+
+            # get a list of fields that we want to check to see if they change
+            # (we want to change model predictions)
+            fields_to_compare = {}
+            for key in new_instances[0].fields:
+                if key not in inputs.keys() and key != name_of_input_field_to_attack:
+                    fields_to_compare[key] = test_instances[0][key]
+
+            current_tokens = new_instances[0][name_of_input_field_to_attack].tokens
```

Comment:
I assume you're going to need to tell mypy to ignore this (and similar things in a bunch of other places), because you're hoping that the given field is actually a `TextField`.  I'd probably explicitly check that and throw an error if it's not, and then here do a `# type: ignore`.  Or you could do something like this:

```python
if not isinstance(new_instance[name_of_input_field_to_attack], TextField):
    raise ValueError("Given input field name was not a TextField")
field_to_attack: TextField = new_instance[name_of_input_field_to_attack]  # type: ignore
```

Then you can reference variables on `field_to_attack` in a type-safe way (and shorten all the code where you access this field).

Also, this argues for keeping `name` in the method argument...  Maybe `input_field_name` and `grad_name`?